wpiformat/wpiformat/licenseupdate.py

```diff
@@ -1,24 +1,24 @@
 """This task updates the license header at the top of the file."""
 
-import os
 import re
 import subprocess
 from datetime import date
+from pathlib import Path
 
 from wpiformat.config import Config
 from wpiformat.task import PipelineTask
 
 
 class LicenseUpdate(PipelineTask):
     @staticmethod
-    def should_process_file(config_file: Config, filename: str) -> bool:
+    def should_process_file(config_file: Config, filename: Path) -> bool:
         license_regex = config_file.regex("licenseUpdateExclude")
 
         return (
             config_file.is_c_file(filename)
             or config_file.is_cpp_file(filename)
-            or filename.endswith(".java")
-        ) and not license_regex.search(filename)
+            or filename.as_posix().endswith(".java")
```

Comment:
```suggestion
            or filename.suffix == ".java"
```
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix